### PR TITLE
[java-spring-boot2]: Fix version update 

### PR DESCRIPTION
**Pull request by ebullient**
This supersedes PR #11

### CODE REVIEW

File: incubator/java-spring-boot2/README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ The Spring Boot 2 stack uses a parent Maven project object model (POM) to manage
 
 The [Spring Developer Tools](https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-devtools.html#using-boot-devtools) module is also included to support automatic restarts and live reloading during development.
 
+*Note:* Maven is provided by the Appsody stack container, allowing you to build, test, and debug your Java application without installing Maven locally. We recommend installing Maven locally for the best IDE experience.
```
Comment (neeraj-laad) [inline incubator/java-spring-boot2/README.md]: Could we make the 'Note' heading bold? The other stacks have had the same issue and wanted to make this a bit more prominent

File: incubator/java-spring-boot2/image/Dockerfile-stack [hunk 2]
```diff
@@ -11,30 +11,27 @@ RUN  apt-get -qq update \
...
 
-WORKDIR "/project"
-
-RUN mkdir -p /project/target && chmod 777 /project/target
+WORKDIR /project/user-app
 
 ENV APPSODY_USER_RUN_AS_LOCAL=true
 
-ENV APPSODY_MOUNTS="src:/project/src;pom.xml:/project/pom.xml;~/.m2/repository:/mvn/repository"
+ENV APPSODY_MOUNTS=".:/project/user-app;~/.m2/repository:/mvn/repository"
```
Comment (neeraj-laad) [inline incubator/java-spring-boot2/image/Dockerfile-stack]: Interestingly the other stack( java-microprofile) went from mounting a single workspace dir to mounting src and pom individually as there were issues with creation of file / folders  that were in the directory that was bind mounted on the host. Were there specific reasons for changing the way the folders are mounted?

File: incubator/java-spring-boot2/image/Dockerfile-stack [hunk 3]
```diff
@@ -11,30 +11,27 @@ RUN  apt-get -qq update \
...
 ENV APPSODY_DEPS=
 
-ENV APPSODY_WATCH_DIR="/project"
-ENV APPSODY_WATCH_IGNORE_DIR="/project/target"
+ENV APPSODY_WATCH_DIR="/project/user-app/src"
+ENV APPSODY_WATCH_IGNORE_DIR=""
 ENV APPSODY_WATCH_REGEX=""
```
Comment (neeraj-laad) [inline incubator/java-spring-boot2/image/Dockerfile-stack]: Would you want to use any filters like only watch for changes to *.java files?

File: incubator/java-spring-boot2/README.md (hunk 1 above)
Comment (ebullient) [reply incubator/java-spring-boot2/README.md]: 👍

File: incubator/java-spring-boot2/image/Dockerfile-stack (hunk 2 above)
Comment (ebullient) [reply incubator/java-spring-boot2/image/Dockerfile-stack]: This is the most significant change, and is what allows the version plugin to run correctly. Remember that the Spring stack runs as user, so any files we create have proper ownership.  I’m not sure the microprofile stack does that yet.. that may have been a different way to fix the issue they were having.

File: incubator/java-spring-boot2/image/Dockerfile-stack (hunk 3 above)
Comment (ebullient) [reply incubator/java-spring-boot2/image/Dockerfile-stack]: no. Spring has other resource files that should also be watched.

File: incubator/java-spring-boot2/image/Dockerfile-stack (hunk 2 above)
Comment (neeraj-laad) [reply incubator/java-spring-boot2/image/Dockerfile-stack]: Yes, run as user is one of the fundamental differences between the 2 stacks.. and maybe it is good to have two different approaches so we can compare and contrast experiences from both before going full on on one.

**Pull request merged**